src/js/helpers.js

```diff
@@ -21,6 +24,22 @@ export function getActivePages(pages, data) {
   return pages.filter(page => isActivePage(page, data));
 }
 
+export function getActiveProperties(activePages) {
+  let allProperties = [];
+  activePages.map(page => {
+    if (page.schema) {
+      allProperties = _.uniq(_.concat(Object.keys(page.schema.properties), allProperties));
```

Comment:
This will still call `_.uniq` once for each page, which could be a lot of times. we can reduce it to just once by pushing all the properties into an array and de-duping once. It's a little easier to read than using a `Set` although it's probably slower, but that probably doesn't matter since it's only called once and even on a large form this shouldn't be horrible.

```js
let allProperties = [];
activePages.forEach(page => allProperties.push(...Object.keys(page.schema.properties));
return _.uniq(allProperties);
```